lib/wallet/src/Cardano/Wallet/Write/Tx/Sign.hs

```diff
@@ -100,30 +108,34 @@ estimateSignedTxSize pparams nWits body =
                     , show feePerByte
                     , "lovelace/byte"
                     ]
+
+
+        -- When updating to a new era, check that the choice of encoding still
+        -- seems right w.r.t the ledger. Types will /probably/ but not
+        -- /necessarily/ protect against calling the wrong function.
         sizeOfTx :: TxSize
-        sizeOfTx = TxSize
-            . fromIntegral
-            . BS.length
-            . serialisedTx
-            $ sealedTxFromCardanoBody body
+        sizeOfTx = fromIntegral @Integer @TxSize $ case era of
+            RecentEraBabbage -> unsignedTx ^. sizeAlonzoTxF
+            RecentEraConway -> unsignedTx ^. sizeAlonzoTxF
```

Comment:
Perhaps we could consider using [`sizeTxF`](https://github.com/input-output-hk/cardano-ledger/blob/2958b712abe939916284fdef49c46704c77d6e87/libs/cardano-ledger-core/src/Cardano/Ledger/Core.hs#L158)?

Reply:
Makes perfect sense - thanks!